[DOCS-1166][DOCS-880][DOCS-1167][DOCS-1168] Update automation docs (#1084)
wandb/docs: content/guides/integrations/huggingface.md

### Turn on model checkpointing 

Using [Artifacts]({{< relref "/guides/core/artifacts/" >}}), you can store up to 100GB of models and datasets for free and then use the Weights & Biases [Registry]({{< relref "/guides/models/registry/" >}}). Using Registry, you can register models to explore and evaluate them, prepare them for staging, or deploy them in your production environment.

> To log your Hugging Face model checkpoints to Artifact, set the `WANDB_LOG_MODEL` environment variable:

Reviewer: **suggestion:** 
```suggestion
To log your Hugging Face model checkpoints to Artifacts, set the `WANDB_LOG_MODEL` environment variable to _one_ of:
```

Author: Edited your edit to avoid "the following".

However, If you pass a [`run_name`](https://huggingface.co/docs/transformers/main/en/main_classes/trainer#transformers.TrainingArguments.run_name) in your `TrainingArguments`, the model will be saved as `model-{run_name}` or `checkpoint-{run_name}`.
{{% /alert %}}

#### W&B Registry

> Once you have logged your checkpoints to Artifacts, you can then register your best model checkpoints and centralize them across your team using the Weights & Biases **[Registry]({{< relref "/guides/models/registry/" >}})**. Using Registry, you can organize your best models by task, manage the lifecycles of models, track and audit the entire ML lifecyle, and [automate]({{< relref "/guides/models/automations/#create-webhook-automation" >}}) downstream actions with webhooks or jobs. 

Reviewer: **suggestion:** 

- I think the convention in most places is to just refer to it as "Registry" (without the definite article)
- Launch is deprecated, but also more generally, automations aren't necessarily intended as specific to just webhooks or Launch actions

```suggestion
#### W&B Registry
Once you have logged your checkpoints to Artifacts, you can then register your best model checkpoints and centralize them across your team with **[Registry]({{< relref "/guides/models/registry/" >}})**. Using Registry, you can organize your best models by task, manage the lifecycles of models, track and audit the entire ML lifecyle, and [automate]({{< relref "/guides/models/automations/#create-webhook-automation" >}}) downstream actions.
```

## Advanced features and FAQs

### How do I save the best model?

> If your code sets `load_best_model_at_end=True` in the `TrainingArguments` you pass to the `Trainer`, W&B saves the best performing model checkpoint to Artifacts.

Reviewer: **suggestion:** 
```suggestion
If you pass `TrainingArguments` with `load_best_model_at_end=True` to your `Trainer`, W&B saves the best performing model checkpoint to Artifacts.
```

Author: Edited your edit to remove the hyphen from "best performing". I personally prefer it with the hyphen too, but we decided as a team to avoid it in cases where Google and MS style guides say it is not needed.
